src/doc/rgw/user-accounts.md

```diff
@@ -0,0 +1,73 @@
+# iam accounts
+
+add a user account primitive for use with IAM REST APIs. this enables end-users to use tools like `aws iam create-user` for self-service management of the users, roles, policies, and access keys in their account without requiring global admin privileges for radosgw-admin commands or radosgw /admin REST APIs
+
+add new IAM APIs for User and AccessKey operations. some Role/RolePolicy/UserPolicy operations are already supported, but require special admin capabilities. once a User or Role are added to an account, that account's root user will be granted full access to them
+
+existing users and roles must be added to an account manually, either by a radosgw-admin command or /admin API, before they can be managed by an account root user
+
+
+## namespacing
+
+rgw supports namespace isolation of users and buckets using the 'tenant' name, such that two tenants A and B can both have buckets named 'example-bucket', or users named 'example-user', that are completely indepedent of one another
+
+there exists no 'tenant' metadata that we can attach policy to, index users or buckets on, or accumulate usage stats for reporting or quota enforcement. features like this will instead be attached to the proposed account metadata
+
+to preserve this namespacing functionality, each account can optionally be associated with a tenant name. all users and roles in that account must have that same tenant name
+
+
+## account metadata
+
+accounts are stored as metadata objects and replicated in multisite by metadata sync
+
+account metadata is created/modified/deleted through a set of radosgw-admin commands ('account create/get/modify/rm') and /admin/account APIs that require admin permissions
+
+the account metadata maintains an index of its users and roles to support API operations like ListUsers and ListRoles
+
+the bucket, user, and role metadata have a new 'account_id' field which refers to their account, if any
```

Comment:
@cbodley could it also be a good idea to add the OIDC Provider as a resource with an ownership link to the account?.

Reply:
thanks, i hadn't planned for this yet. Topic metadata also needs to be adapted for account ownership. i added these to the TODO list in https://github.com/ceph/ceph/pull/54333

Reply:
Thanks